pkg/cmd/init/exec.go

```diff
@@ -192,12 +195,40 @@ func (o *Options) run() error {
 		}
 	}
 
-	fmt.Printf("The multicluster hub control plane has been initialized successfully!\n\n"+
-		"You can now register cluster(s) to the hub control plane. Log onto those cluster(s) and run the following command:\n\n"+
-		"    %s --cluster-name <cluster_name>\n\n"+
-		"Replace <cluster_name> with a cluster name of your choice. For example, cluster1.\n\n",
-		cmd,
-	)
+	if o.jsonOutput {
+		err := WriteJsonOutput(os.Stdout, token, restConfig.Host)
+		if err != nil {
+			return err
+		}
+	} else {
+		fmt.Printf("The multicluster hub control plane has been initialized successfully!\n\n"+
+			"You can now register cluster(s) to the hub control plane. Log onto those cluster(s) and run the following command:\n\n"+
+			"    %s --cluster-name <cluster_name>\n\n"+
+			"Replace <cluster_name> with a cluster name of your choice. For example, cluster1.\n\n",
+			cmd,
+		)
+	}
 
 	return apply.WriteOutput(o.outputFile, output)
 }
+
+func WriteJsonOutput(w io.Writer, token, host string) error {
```

Comment:
This works only for writing hub info, but it will be more useful it will be:

    func WriteJsonOutput(w io.Writer, v any) error {


so when we add json output to other command it can use the same helper to write json.